addons/calendar/models/calendar_event.py

```diff
@@ -446,7 +446,17 @@ def create(self, vals_list):
         model_ids = list(filter(None, {values.get('res_model_id', defaults.get('res_model_id')) for values in vals_list}))
         model_name = defaults.get('res_model')
         valid_activity_model_ids = model_name and self.env[model_name].sudo().browse(model_ids).filtered(lambda m: 'activity_ids' in m).ids or []
-        if meeting_activity_type and not defaults.get('activity_ids'):
+
+        # if user is creating an event for an activity that already has one, create a second activity
+        existing_event = False
+        default_activity_ids_commands = defaults.get('activity_ids', [])
+        if len(default_activity_ids_commands) == 1 and default_activity_ids_commands[0][0] == Command.SET:
+            default_activity_ids = self.env['mail.activity'].browse(default_activity_ids_commands[0][2]).exists()
+            if len(default_activity_ids) == 1 and default_activity_ids.calendar_event_id:
+                existing_event = default_activity_ids.calendar_event_id
+                meeting_activity_type = default_activity_ids.activity_type_id.filtered(lambda at: at.category == 'meeting') or meeting_activity_type
+
+        if meeting_activity_type and (not defaults.get('activity_ids') or existing_event):
```

Comment:
Hello, thanks for your PR. Honestly, I find this whole block fragile. I think it may break. Parsing ORM commands is often a bad idea. Moreover the create method of the calendar event model is already huge and difficult to read. This block don't improve readability.  Couldn't you you check the length of vals_list, the length of activities and  decide if you need to create more ? maybe externalize the activity block to improve readability ?

Reply:
Thank you for the feedback! I will work towards an improved solution. I believe that checking the length of the vals_list would not work because under this use case, the create() method is called separately for each calendar event creation as opposed to being batched so the length of vals_list will be 1 each time.